# Ki-Kare Uyum Testi (Chi-Square Goodness of Fit Test)
Ki-Kare Uyum Testi, **kategorik bir değişkenin** gözlenen dağılımının, **beklenen/iddia edilen bir dağılıma uyup uymadığını** test eder. Tüm şimdiye kadarki testlerden farklı olarak, artık ortalamalarla değil, **kategorilerin frekanslarıyla (sayımlarıyla)** çalışıyoruz.

## Ne Zaman Kullanılır?
Bir kategorik değişkenin (örn: "Hangi kanaldan sipariş geldi") gözlenen dağılımını, **teorik/beklenen bir dağılımla** karşılaştırmak istediğinde. Örnek: "Şirket, siparişlerin kanallara EŞİT dağıldığını iddia ediyor (her kanaldan %25). Gerçek veri bunu destekliyor mu?"

## Hipotezler
- **H0:** Gözlenen dağılım, beklenen dağılıma uyar
- **H1:** Gözlenen dağılım, beklenen dağılıma uymaz

## Test İstatistiği Formülü
$$\chi^2 = \sum \frac{(O_i - E_i)^2}{E_i}$$

- $O_i$ (Observed) → her kategoride **gözlenen** frekans
- $E_i$ (Expected) → her kategoride **beklenen** frekans (H0'a göre)

**Sezgisel anlamı:** Her kategoride, gözlenenin beklenenden ne kadar saptığını (karesini alarak, negatifleri pozitif yaparak — Konu 11'deki std sapma mantığının aynısı) ölçüp topluyoruz. Sapma ne kadar büyükse, Ki-Kare değeri o kadar büyür.

## Örnek Senaryo
Bir şirket "siparişlerin 4 kanala (Web, Mobil Uygulama, Telefon, Mağaza) EŞİT dağıldığını" (%25'er) iddia ediyor. 400 siparişlik bir örneklemde: Web=130, Mobil=110, Telefon=90, Mağaza=70 gözlendi. Bu, "eşit dağılım" iddiasını destekliyor mu?

## Python'da Kullanımı
```python
from scipy.stats import chisquare

gozlenen = [130, 110, 90, 70]
beklenen = [100, 100, 100, 100]  # 400/4 = eşit dağılımda her kategori 100

chi2_stat, p_degeri = chisquare(f_obs=gozlenen, f_exp=beklenen)
```

In [1]:
# Bir online oyun şirketi, oyunda kullanılan sanal zarın adil olduğunu iddia ediyor (yani her yüzün gelme olasılığı eşit, 1/6). Şüpheli bir oyuncu, zarı 300 kere attırıyor ve şu sonuçları kaydediyor:
# 1 geldi: 42 kere
# 2 geldi: 55 kere
# 3 geldi: 38 kere
# 4 geldi: 65 kere
# 5 geldi: 48 kere
# 6 geldi: 52 kere

# H0: Gözlenen dağılım, beklenen dağılıma uyar.
# H1: Gözlenen dağılım, beklenen dağılıma uymaz.

from scipy.stats import chisquare
import numpy as np
import pandas as pd

pd.set_option('display.float_format', lambda x: f'{x:.8f}')
gozlenen = np.array([42, 55, 38, 65, 48, 52])
beklenen = np.array([50, 50, 50, 50, 50, 50])

chi2_stat, p_degeri = chisquare(f_obs = gozlenen, f_exp = beklenen)
alpha = 0.05
print(f'Ki-Kare istatistiği: {chi2_stat}')
print(f'P-değeri: {p_degeri}')
if p_degeri < alpha:
    print('H0 reddedilir, gözlenen dağılım beklenen dağılıma uymuyor.')
else:
    print('H0 reddedilemez, gözlenen dağılım beklenen dağılıma uyuyor.')

Ki-Kare istatistiği: 9.32
P-değeri: 0.09696125366335304
H0 reddedilemez, gözlenen dağılım beklenen dağılıma uyuyor.


### Sonuç
300 kere zar atışında gözlenen frekanslar ile adil bir zarda beklenen frekansları (her yüz için 50) karşılaştırmak amacıyla Ki-Kare Uyum 
Testi uyguladık. Test sonucuna göre (χ²=9.32, p=0.097), gözlenen dağılımın beklenen (eşit) dağılımdan anlamlı şekilde sapmadığı, yani 
elimizdeki verinin zarın hileli olduğuna dair yeterli kanıt sunmadığı sonucuna varılmıştır.